molgenis-data-mapper/src/test/java/org/molgenis/data/mapper/service/impl/MappingServiceImplTest.java

```diff
@@ -186,367 +242,268 @@ public void testGetMappingProject()
+		targetMeta.addAttribute(attrMetaFactory.create().setName("source"));
+
+		when(addEntityRepo.getEntityMetaData()).thenReturn(targetMeta);
+		when(metaDataService.addEntityMeta(argThat(new ArgumentMatcher<EntityMetaData>()
+		{
+			@Override
+			public boolean matches(Object obj)
+			{
+				return obj instanceof EntityMetaData && ((EntityMetaData) obj).getName().equals(entityName);
+			}
+		}))).thenReturn(addEntityRepo);
 
-		MappingProject mappingProject = mappingService.addMappingProject("Test123", user, HOP_ENTITY);
-
-		// now add new source
-		EntityMapping mapping = mappingProject.getMappingTarget(HOP_ENTITY).addSource(geneMetaData);
-		mappingService.updateMappingProject(mappingProject);
+		// add an entity to the source
+		List<Entity> sourceGeneEntities = newArrayList();
+		List<Entity> expectedEntities = newArrayList();
+		createEntities(targetMeta, sourceGeneEntities, expectedEntities);
+
+		Query<Entity> addEntityQ = mock(Query.class);
+		when(addEntityRepo.query()).thenReturn(addEntityQ);
+		when(addEntityQ.eq("source", SOURCE_GENE_ENTITY)).thenReturn(addEntityQ);
+		when(addEntityQ.findAll()).thenAnswer(new Answer<Stream<Entity>>()
+		{
+			@Override
+			public Stream<Entity> answer(InvocationOnMock invocation) throws Throwable
+			{
+				return Stream.empty();
+			}
+		});
 
-		MappingProject retrieved = mappingService.getMappingProject(mappingProject.getIdentifier());
-		assertEquals(retrieved, mappingProject);
+		when(geneRepo.iterator()).thenAnswer(new Answer<Iterator<Entity>>()
+		{
+			@Override
+			public Iterator<Entity> answer(InvocationOnMock invocation) throws Throwable
+			{
+				return sourceGeneEntities.iterator();
+			}
+		});
 
-		assertEquals(retrieved.getMappingTarget(HOP_ENTITY).getMappingForSource(GENE_ENTITY), mapping);
-	}
+		// make project and apply mappings once
+		MappingProject project = createMappingProjectWithMappings();
 
-	//@Test
-	public void testAddExistingSource()
-	{
-		when(idGenerator.generateId()).thenReturn(uuidGenerator.generateId());
+		// apply mapping again
+		String generatedEntityName = mappingService
+				.applyMappings(project.getMappingTarget(TARGET_HOP_ENTITY), entityName);
+		assertEquals(generatedEntityName, entityName);
 
-		MappingProject mappingProject = mappingService.addMappingProject("Test123", user, HOP_ENTITY);
-		mappingProject.getMappingTarget(HOP_ENTITY).addSource(geneMetaData);
+		ArgumentCaptor<Stream<Entity>> streamCaptor = forClass((Class) Stream.class);
+		verify(addEntityRepo).add(streamCaptor.capture());
 
-		mappingService.updateMappingProject(mappingProject);
-		MappingProject retrieved = mappingService.getMappingProject(mappingProject.getIdentifier());
-		try
-		{
-			retrieved.getMappingTarget(HOP_ENTITY).addSource(geneMetaData);
-			fail("Expected exception");
-		}
-		catch (IllegalStateException ignored)
-		{
-		}
-	}
+		assertTrue(EntityUtils.entitiesEquals(streamCaptor.getValue(), expectedEntities.stream()));
```

Comment:
entitiesEquals(Stream<Entity>, Stream<Entity>) does not exist in EntityUtils. Forgot to commit?